### DISCUSSION TITLE
Database UML

### DISCUSSION THREAD

**Discussion by dirkronsmans** [opendcim/openDCIM#1563] [Q&A] [answered]
Hi,
For an internal project we're looking to pull data out of OpenDCIM if possible directly from the database.
To do this as efficiently as possible i'm looking for some kind of database model/uml diagram that can guide me in what data to find where.

Does such a thing exist publicly or could someone guide me to where I can find this?

Thanks!

**Comment by samilliken** [comment, marked as answer]
No, I had one at one time but it got way out of sync from reality.

Scott
________________________________
From: Dirk R. ***@***.***>
Sent: Monday, September 30, 2024 4:30 AM
To: opendcim/openDCIM ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [opendcim/openDCIM] Database UML (Discussion #1563)


Hi,
For an internal project we're looking to pull data out of OpenDCIM if possible directly from the database.
To do this as efficiently as possible i'm looking for some kind of database model/uml diagram that can guide me in what data to find where.

Does such a thing exist publicly or could someone guide me to where I can find this?

Thanks!

**Comment by alex001x** [comment]
hello,

i use MySQL Workbench
https://dev.mysql.com/downloads/workbench/

export mysql file without datas,  only structure table

**Reply by wilpig** [reply]
While Scott did not say it directly I will say it, we do not recommend pulling data from the database directly.  Use the API.  There will be relations in the database that may not be evident and we handle those cases through the data models to maintain the integrity.
